modules/RTC/TraceablePeerConnection.js

```diff
@@ -1688,19 +1688,15 @@ TraceablePeerConnection.prototype._mungeCodecOrder = function(description) {
     }
 
     for (const mLine of mLines) {
-        if (this.codecSettings.disabled) {
-            SDPUtil.stripCodec(mLine, this.codecSettings.disabled);
-        }
-
-        if (this.codecSettings.preferred) {
-            SDPUtil.preferCodec(mLine, this.codecSettings.preferred);
+        for (const codec of this.codecSettings.codecList.slice().reverse()) {
+            SDPUtil.preferCodec(mLine, codec);
 
             // Strip the high profile H264 codecs on mobile clients for p2p connection. High profile codecs give better
             // quality at the expense of higher load which we do not want on mobile clients. Jicofo offers only the
             // baseline code for the jvb connection and therefore this is not needed for jvb connection.
             // TODO - add check for mobile browsers once js-utils provides that check.
-            if (this.codecSettings.preferred === CodecMimeType.H264 && browser.isReactNative() && this.isP2P) {
-                SDPUtil.stripCodec(mLine, this.codecSettings.preferred, true /* high profile */);
+            if (codec === CodecMimeType.H264 && browser.isReactNative() && this.isP2P) {
```

Comment:
Why not remove the TODO while we're at it, and change the check to isMobileDevice?

Reply:
Yes, didn't realize that! will do